### ISSUE TITLE
Cursor 

### ISSUE THREAD

**Issue by HernandoR** [zsh-users/zsh-syntax-highlighting#789]
Cursor  sways with this plugin, 

I currently using MSYS2 in windows 10 with oh-my-zsh.

However, the Cursor will automatically change to the head of the line and then run back where it should be every time I tap the first few letters.

Such thing only happens with this plugin opened, and when I using windows terminal, If I use MSYS2 directly, it does not occur or I haven't noticed.

**Comment by wight554** [comment]
Seems like it's related to ConPTY
winpty/mintty are not affected
I faced same issue in any terminal with conpty backend (WIndows terminal/Alacritty/VS Code terminal)

**Comment by wight554** [comment]
There's related issue in conpty itself
https://github.com/microsoft/terminal/issues/364
But seems like it's dead

**Comment by danielshahaf** [comment]
@wight554 Thanks!

@HernandoR Does the referenced issue describe your setup too?

If so, we should probably close this ticket as an upstream bug?

**Comment by wight554** [comment]
Uh that's an actual nightmare
MS guys lost that ticket perhaps, it's been 2 years since the last update
If nothing can be don't from plugin side, then there's no choice other than closing ticket

**Comment by danielshahaf** [comment]
zsh-syntax-highlighting doesn't do anything related to positioning the cursor. zle (part of zsh) might, so you might try asking zsh-users@ with a minimal reproducer (one that's written in `zsh -f` and doesn't `source` anything).

**Comment by wight554** [comment]
Well it happens only with this (or fsyh) plugin enabled
Issue I linked higher has repro steps

**Comment by danielshahaf** [comment]
> Well it happens only with this (or fsyh) plugin enabled
> Issue I linked higher has repro steps

I've seen that, but I stand by my recommendation to come up with a _minimal_ reproducer, as outlined above — or, at least, as minimal as practicable.

**Comment by chrstophr-wltrs** [comment]
I'm having this same issue, but I thought I would contribute with gif demonstrations, since this problem is kind of hard to describe. 

Mintty:  [with z-s-h](https://gfycat.com/repentantanxiousgoshawk), [no z-s-h](https://gfycat.com/quickleanjohndory)
WinTerm: [with z-s-h](https://gfycat.com/handydesertedcob), [no z-s-h](https://gfycat.com/messyagitatedgoldenretriever)

I can confirm that the WinTerm behavior is identical in VS Code's integrated terminal. 

This is an _extremely_ niche use case (how many people are using zsh-syntax-highlighting on Windows?), so it's probably not a very high priority, and it's just _barely_ noticeable enough to be distracting, but I thought I'd try to document the behavior, in case that's at all helpful.

**Version Information**
z-s-h: 0.8.0-alpha2-dev
zsh: 5.8 (x86_64-pc-cygwin)
Cygwin: 3.3.4
WinTerm: 1.11.3471.0
Mintty: 3.6.0
VS Code: 1.66.0

**Comment by danielshahaf** [comment]
@chrstophr-wltrs Thanks, that's very helpful.  Could you upload the gifs as attachments to your next comment, please?  See instructions at the bottom of the textarea comments are entered in.

**Comment by chrstophr-wltrs** [comment]
Yes, absolutely! Sorry about that, I've updated the links in my first post to point to images hosted on Github instead of gfycat.

**Comment by helmesjo** [comment]
Just to shine some light on this: Has there been any progress or any revelations as to how it can be worked around? [This example](https://user-images.githubusercontent.com/29496790/163515630-40e71160-9957-4d5a-826e-4ecdbfccec28.gif) by @chrstophr-wltrs is exactly what I'm seeing (with alacritty + msys2 (UCRT) + zsh FWIW).
